### PR TITLE
Note about SHA-1 values in the HEAD file

### PR DESCRIPTION
**Pull request by SiarheiBobryk**
I'd like to add some notes about the HEAD file content which I think
should be in this book section.

Unfortunately I'm not English native speaker so please correct me if I
did grammar or other mistakes.

### CODE REVIEW

File: book/10-git-internals/sections/refs.asc [hunk 1]
```diff
@@ -70,8 +70,12 @@ When you run commands like `git branch <branch>`, Git basically runs that `updat
 The question now is, when you run `git branch <branch>`, how does Git know the SHA-1 of the last commit?
 The answer is the HEAD file.
 
-The HEAD file is a symbolic reference to the branch you're currently on.
-By symbolic reference, we mean that unlike a normal reference, it doesn't generally contain a SHA-1 value but rather a pointer to another reference.
+Usually the HEAD file is a symbolic reference to the branch you're currently on.
+By symbolic reference, we mean that unlike a normal reference, it contains a pointer to another reference.
+
+However in some rare cases the HEAD file may contain a SHA-1 value of a git object (commit or tag).
```
Comment (ben) [inline book/10-git-internals/sections/refs.asc]: I have a small suggestion regarding wording:

```suggestion
However in some rare cases the HEAD file may contain the SHA-1 value of a git object.
```

Also, I'm not sure you can get HEAD to point to a tag object directly:

```
--- /tmp/foo ‹master› » git log --oneline --graph --decorate | cat
* 9bf8226 (HEAD -> master) second
* d73043b (tag: tag1) initial
--- /tmp/foo ‹master› » git show tag1 | cat
tag tag1
Tagger: Ben Straub <ben@straub.cc>
Date:   Sat Jun 1 08:36:58 2019 -0700

tag

commit d73043b64f912984f5c5577abf265587407be75b
Author: Ben Straub <ben@straub.cc>
Date:   Sat Jun 1 08:36:01 2019 -0700

    initial
--- /tmp/foo ‹master› » cat .git/refs/tags/tag1
4cbdc2bbff77b80bb2963aae1e80b5b311d5a3ff
--- /tmp/foo ‹master› » git checkout 4cbdc2bbff77b80bb2963aae1e80b5b311d5a3ff
Note: checking out '4cbdc2bbff77b80bb2963aae1e80b5b311d5a3ff'.

You are in 'detached HEAD' state. You can look around, make experimental
changes and commit them, and you can discard any commits you make in this
state without impacting any branches by performing another checkout.

If you want to create a new branch to retain commits you create, you may
do so (now or later) by using -b with the checkout command again. Example:

  git checkout -b <new-branch-name>

HEAD is now at d73043b initial
```

When I tried to check out the tag object, Git instead took me to a detached head on the commit the tag pointed to. So we can leave out the "commit or tag" parenthetical; HEAD is either symbolic or it points to a commit.

File: book/10-git-internals/sections/refs.asc [hunk 2]
```diff
@@ -70,8 +70,12 @@ When you run commands like `git branch <branch>`, Git basically runs that `updat
...
+You can achieve it by checking out to a specific tag, commit or remote branch which puts your repository in https://git-scm.com/docs/git-checkout#_detached_head["detached HEAD"] state.
```
Comment (ben) [inline book/10-git-internals/sections/refs.asc]: Another small suggestion:

```suggestion
This happens when you checkout a tag, commit, or remote branch, which puts your repository in https://git-scm.com/docs/git-checkout#_detached_head["detached HEAD"] state.
```

File: book/10-git-internals/sections/refs.asc (hunk 1 above)
Comment (SiarheiBobryk) [reply book/10-git-internals/sections/refs.asc]: Yep. My fault. No directly of course.
So
- by checking out to lightweight tag Git takes the commit the tag pointed to
- by checking out to annotated tag Git takes the commit the tag object pointed to

Comment (ben) [reply book/10-git-internals/sections/refs.asc]: Correct.

Comment (SiarheiBobryk) [reply book/10-git-internals/sections/refs.asc]: fixed

File: book/10-git-internals/sections/refs.asc (hunk 2 above)
Comment (SiarheiBobryk) [reply book/10-git-internals/sections/refs.asc]: Fixed!

**Comment by SiarheiBobryk** [comment]
So much appreciate Ben!
Thanks again for your suggestions.

**Comment by ben** [comment]
✨

**Pull request merged**